bofire/surrogates/mlp.py

```diff
@@ -109,40 +125,48 @@ def forward(self, X: Tensor):
     @property
     def num_outputs(self) -> int:
         r"""The number of outputs of the model."""
-        return self.mlps[0].layers[-1].out_features  # type: ignore
+        return self.mlps[0].layers[-2].out_features  # type: ignore
 
 
 def fit_mlp(
     mlp: MLP,
-    dataset: RegressionDataSet,
+    dataset: MLPDataset,
     batch_size: int = 10,
     n_epoches: int = 200,
     lr: float = 1e-4,
     shuffle: bool = True,
     weight_decay: float = 0.0,
+    loss_function=nn.L1Loss,  # type: ignore
 ):
-    """Fit a MLP for regression to a dataset.
+    """Fit a MLP to a dataset.
 
     Args:
         mlp (MLP): The MLP that should be fitted.
-        dataset (RegressionDataSet): The data that should be fitted
+        dataset (MLPDataset): The data that should be fitted
         batch_size (int, optional): Batch size. Defaults to 10.
         n_epoches (int, optional): Number of training epoches. Defaults to 200.
         lr (float, optional): Initial learning rate. Defaults to 1e-4.
         shuffle (bool, optional): Whereas the batches should be shuffled. Defaults to True.
         weight_decay (float, optional): Weight decay (L2 regularization). Defaults to 0.0 (no regularization).
+        loss_function (Loss function, NOT Optional): Loss function specified by the problem type. Defaults to L1 loss for regression problems.
     """
     mlp.train()
     train_loader = DataLoader(dataset=dataset, batch_size=batch_size, shuffle=shuffle)
-    loss_function = nn.L1Loss()
     optimizer = torch.optim.Adam(mlp.parameters(), lr=lr, weight_decay=weight_decay)
+    loss_function = loss_function()
     for _ in range(n_epoches):
         current_loss = 0.0
         for data in train_loader:
             # Get and prepare inputs
             inputs, targets = data
-            if len(targets.shape) == 1:
+            if isinstance(loss_function, nn.CrossEntropyLoss):
+                targets = targets.flatten().long()
+            elif len(targets.shape) == 1 and not isinstance(
+                loss_function, nn.CrossEntropyLoss
+            ):
                 targets = targets.reshape((targets.shape[0], 1))
+            else:
+                pass
```

Comment:
An error message? Or we can just remove the else if we do nothing.

Reply:
I will remove